codice/ddf
# DDF-4658 Adding toggle for spellcheck feature

#### What does this PR do?
Adds the ability to turn on/off the spellcheck feature in search settings. 
Also adds the ability to turn on/off the spellcheck in the admin settings (turning it off in the admin under Catalog UI Search means that the spellcheck toggle will not even show up inside intrigue)

#### Who is reviewing it? 
@bellcc 
@derekwilhelm 
@andrewkfiedler 
@tbatie 

#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler
@tbatie

#### How should this be tested?
1. Upload/ingest data
2. Check that you are able to toggle hit on/off for spellcheck in Default Search Settings and in Query Search Settings without errors.
3. Go to admin config under Catalog UI Search and turn off Spellcheck.
4. Go back to intrigue and ensure that the spellcheck option no longer appears in Default Search Settings nor in Query Search Settings

#### Any background context you want to provide?
#### What are the relevant tickets?
For GH Issues:
Fixes: #4658 

#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
